openstack/sharedfilesystems/v2/shareaccessrules/results.go

```diff
@@ -0,0 +1,65 @@
+package shareaccessrules
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// ShareAccess contains information associated with an OpenStack share access rule.
+type ShareAccess struct {
+	// The UUID of the share to which you are granted or denied access.
+	ShareID string `json:"share_id"`
+	// The date and time stamp when the resource was created within the service’s database.
+	CreatedAt time.Time `json:"-"`
+	// The date and time stamp when the resource was last updated within the service’s database.
+	UpdatedAt time.Time `json:"-"`
+	// The access rule type.
+	AccessType string `json:"access_type"`
+	// The value that defines the access. The back end grants or denies the access to it.
+	AccessTo string `json:"access_to"`
+	// The access credential of the entity granted share access.
+	AccessKey string `json:"access_key"`
+	// The state of the access rule.
+	State string `json:"state"`
+	// The access level to the share.
+	AccessLevel string `json:"access_level"`
+	// The access rule ID.
+	ID string `json:"id"`
+	// Access rule metadata.
+	Metadata map[string]interface{} `json:"metadata"`
+}
+
+func (r *ShareAccess) UnmarshalJSON(b []byte) error {
+	type tmp ShareAccess
+	var s struct {
+		tmp
+		CreatedAt gophercloud.JSONRFC3339MilliNoZ `json:"created_at"`
```

Comment:
According to [the doc](https://docs.openstack.org/api-ref/shared-file-system/?expanded=list-access-rules-deprecated-detail#id610), the date is returned in ISO 8601 format. Was JSONRFC3339MilliNoZ perhaps an old date format used by Manila?

Reply:
It seems RFC3339 offers ISO 8601 interoperability: https://www.rfc-editor.org/rfc/rfc3339. Also other structs seem to use this JSONRFC3339MilliNoZ where the docs mention ISO 8601... I'm not an expert in this area

Reply:
I don't get the question perhaps, all date and timestamp fields from openstack services use this datatype within the gophercloud SDK:

https://github.com/gophercloud/gophercloud/blob/513734676e6495f6fec60e7aaf1f86f1ce807428/results.go#L308

Reply:
I've reported https://github.com/gophercloud/gophercloud/issues/2507 for the date parsing issue if anyone wants to take a crack at it.